readthedocs-custom-steps/src/readthedocs_custom_steps.py

```diff
@@ -29,8 +29,25 @@ def main():
     print('readthedocs-custom-steps', __version__)
     return
 
-  filename = '.readthedocs-custom-steps.yml'
-  if os.path.isfile(filename):
+  def recursive_find_filename(fname, directory='.'):
+    for current_path, directories, files in os.walk(directory):
+      if os.path.split(current_path)[-1] == 'test':
+        continue
+      elif fname in files:
+        return os.path.join(directory, fname)
+      elif directories == '':
```

Comment:
`os.walk()` maybe returns an empty sequence for `directories`, but that sequence surely is not an empty string. The whole check can be omitted as it wouldn't change the behaviour with the for-loop in the `else:` block.